modules/backendscheduler/backendscheduler.go

```diff
@@ -82,123 +114,98 @@ func (s *BackendScheduler) starting(ctx context.Context) error {
 		return fmt.Errorf("failed to load work cache: %w", err)
 	}
 
+	// Start providers and collect job channels
+	s.mergedJobs = make(chan *work.Job, len(s.providers))
+
+	wg := sync.WaitGroup{}
+
+	for i := range s.providers {
+		s.providers[i].jobs = s.providers[i].provider.Start(ctx)
+
+		wg.Add(1)
+		// Start a goroutine to forward jobs from each provider to the merged channel
+		go func(jobs <-chan *work.Job) {
+			defer wg.Done()
+
+			var job *work.Job
+
+			for {
```

Comment:
suggested cleanup:

```
for {
  select {
    case job = <-jobs:
    case <-ctx.Done():
      return
   }
   
   select {
     case s.mergedJobs <- job:
     case <-ctx.Done():
       return
   } 
}
```

would remove the timer and the job nil handling and remove the potential to spin tightly if s.mergedJobs is full